Core/include/Acts/Material/GridSurfaceMaterialFactory.hpp

```diff
@@ -0,0 +1,210 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Material/GridSurfaceMaterial.hpp"
+#include "Acts/Material/MaterialSlab.hpp"
+#include "Acts/Utilities/GridAccessHelpers.hpp"
+#include "Acts/Utilities/ProtoAxis.hpp"
+
+#include <memory>
+#include <vector>
+
+namespace Acts::GridSurfaceMaterialFactory {
+
+/// Create and fill from a single proto axis
+///
+/// @param pAxis the type of the ProtoAxis
+/// @param materialAccessor the material accessor
+/// @param boundToGridLocal the delegate from bound to grid local frame
+/// @param globalToGridLocal the delegate from global into grid local frame
+/// @param payload the grid payload (material slab / indices)
+///
+/// @return a unique pointer to the surface material
+template <typename material_accessor_t>
+std::unique_ptr<
+    IGridSurfaceMaterial<typename material_accessor_t::grid_value_type>>
+create1D(
```

Comment:
Very nice! Have you considered / tried moving these templates also over to the `.cpp` file? If it's only ever instantiated there it should be possible.

Reply:
At the moment that's true, I just wanted to leave the opportunity for someone who wanted to introduce a new material_accessor ...